### PR TITLE
Implement and export some `WasiFile`s

### PR DESCRIPTION
**Pull request by MarkMcCaskey**
I wasn't sure how to push commits to the fork, so I'm spinning #1844 off here

Just need to resolve the issue with wasi tests now, looks like we don't support providing input for wasi tests so this won't be able to pass tests yet even if we moved it over to wasmerio/wasi-tests.

Will look into moving the example somewhere else or adding support for provided input via stdin

### CODE REVIEW

**Comment by MarkMcCaskey** [comment]
bors try

File: tests/wasi-wast/wasi/unstable/pipe_reverse.wast [hunk 1]
```diff
@@ -0,0 +1,7 @@
+;; This file was generated by https://github.com/wasmerio/wasi-tests
+
+(wasi_test "pipe_reverse.wasm"
+  (assert_return (i64.const 0))
+  (provide_stdin "Hello, world!")
+  (assert_stdout "!dlrow ,olleH\n")
```
Comment (syrusakbary) [inline tests/wasi-wast/wasi/unstable/pipe_reverse.wast:6]: Nice!

File: tests/lib/wast/src/wasi_wast.rs [hunk 2]
```diff
@@ -21,6 +21,7 @@ pub struct WasiTest<'a> {
     mapped_dirs: Vec<(&'a str, &'a str)>,
     temp_dirs: Vec<&'a str>,
     assert_return: Option<AssertReturn>,
+    provide_stdin: Option<ProvideStdin<'a>>,
```
Comment (syrusakbary) [inline tests/lib/wast/src/wasi_wast.rs:24]: I will name it `stdin` rather than `provide_stdin` (what can be stdin otherwise?)

Comment (MarkMcCaskey) [reply tests/lib/wast/src/wasi_wast.rs:24]: Fair point!

My thought was just making it symmetrical with `assert_stdout`, I can change it to `stdin` though

File: tests/lib/wast/src/wasi_wast.rs [hunk 3]
```diff
@@ -179,6 +190,7 @@ mod wasi_kw {
     wast::custom_keyword!(map_dirs);
     wast::custom_keyword!(temp_dirs);
     wast::custom_keyword!(assert_return);
+    wast::custom_keyword!(provide_stdin);
```
Comment (syrusakbary) [inline tests/lib/wast/src/wasi_wast.rs:193]: `stdin` ?

**Review by syrusakbary — APPROVED** [review]

**Pull request merged**